src/components/StateSelector.tsx

```diff
@@ -0,0 +1,103 @@
+import {useIsFocused} from '@react-navigation/native';
+import {CONST as COMMON_CONST} from 'expensify-common/lib/CONST';
+import React, {useEffect, useRef} from 'react';
+import type {ForwardedRef} from 'react';
+import {View} from 'react-native';
+import useGeographicalStateFromRoute from '@hooks/useGeographicalStateFromRoute';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import type {MaybePhraseKey} from '@libs/Localize';
+import Navigation from '@libs/Navigation/Navigation';
+import ROUTES from '@src/ROUTES';
+import FormHelpMessage from './FormHelpMessage';
+import type {MenuItemProps} from './MenuItem';
+import MenuItemWithTopDescription from './MenuItemWithTopDescription';
+
+type State = keyof typeof COMMON_CONST.STATES;
+
+type StateSelectorProps = {
+    /** Form error text. e.g when no state is selected */
+    errorText?: MaybePhraseKey;
+
+    /** Current selected state  */
+    value?: State | '';
+
+    /** Callback to call when the input changes */
+    onInputChange?: (value: string) => void;
+
+    /** Label to display on field */
+    label?: string;
+
+    /** Any additional styles to apply */
+    wrapperStyle?: MenuItemProps['wrapperStyle'];
+
+    /** Callback to call when the picker modal is dismissed */
+    onBlur?: () => void;
+
+    /** object to get route details from */
+    stateSelectorRoute?: typeof ROUTES.SETTINGS_ADDRESS_STATE | typeof ROUTES.MONEY_REQUEST_STATE_SELECTOR;
+};
+
+function StateSelector(
+    {errorText, onBlur, value: stateCode, label, onInputChange, wrapperStyle, stateSelectorRoute = ROUTES.SETTINGS_ADDRESS_STATE}: StateSelectorProps,
+    ref: ForwardedRef<View>,
+) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const stateFromUrl = useGeographicalStateFromRoute();
+
+    const didOpenStateSelector = useRef(false);
+    const isFocused = useIsFocused();
+
+    useEffect(() => {
+        if (isFocused && didOpenStateSelector.current) {
+            didOpenStateSelector.current = false;
+            if (!stateFromUrl) {
+                onBlur?.();
+            }
+        }
+
+        if (!stateFromUrl) {
+            return;
+        }
+
+        // This will cause the form to revalidate and remove any error related to country name
+        if (onInputChange) {
+            onInputChange(stateFromUrl);
+        }
+
+        Navigation.setParams({state: undefined});
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [stateFromUrl, onBlur, isFocused]);
+
+    const title = stateCode && Object.keys(COMMON_CONST.STATES).includes(stateCode) ? translate(`allStates.${stateCode}.stateName`) : '';
```

Comment:
Yes, it won't work if not used within form. But I suppose since it in an input component, it will always be used inside a form. Do you want it to be more independent, generic, futureproof, and have its own state?

Reply:
Currently it is not used without a `Form `and If it is used as a standalone component, it can always be wrapped around a `Form `like `IncorporationStateBusiness`. Or the devs can modify the code as per their needs. For current uses and its intended purpose, do you think it still needs a local state?
![image](https://github.com/Expensify/App/assets/65986357/2880a200-b24e-4b9e-90cd-90ed0c42bab6)

Reply:
Current [StatePicker](https://github.com/Expensify/App/blob/main/src/components/StatePicker/index.tsx) too doesnt have local state. Since we're just mimicking its functionality, do you still think we need to worry about local state?

Reply:
Done